src/main/java/cn/nukkit/block/BlockRail.java

```diff
@@ -142,8 +142,26 @@ public boolean place(Item item, Block block, Block target, BlockFace face, doubl
         if (!isAbstract()) {
             level.scheduleUpdate(this, this, 0);
         }
+        
         return true;
     }
+    
+    private boolean checkCanBePlace(Block check) {
+        if (check == null) {
+            return false;
+        }
+        if (!check.isTransparent()) {
+            return true;
+        }
+        if (check instanceof BlockHopper) {
+            return true;
+        } else if (check instanceof BlockSlab) {
+            return (check.getDamage() & 0x8) == 0x8;
+        } else if (check instanceof BlockStairs) {
+            return (check.getDamage() & 0x4) == 0x4;
+        }
```

Comment:
Please, use the property accessors like `((BlockSlab)check).isOnTop()` and `((BlockStairs)check).isUpsideDown()` instead of magic values

Reply:
Thank you